examples/camera/camera_main.c

```diff
@@ -525,74 +536,182 @@ int main(int argc, FAR char *argv[])
+   *    After 10 seconds, this application will be finished.
+   *
+   * Notice:
+   *    If capture_num is set '0', state will stay APP_STATE_BEFORE_CAPTURE.
+   */
 
-  RESET_INITIAL_TIME(then);
+  app_state = APP_STATE_BEFORE_CAPTURE;
 
-  /* =====  Main Loop  ===== */
+  /* Show this application behavior. */
 
-  while (1)
+  if (capture_num == 0)
     {
-      printf("Start capturing...\n");
-      ret = start_stillcapture(v_fd, capture_type);
-      if (ret != OK)
-        {
-          goto exit_this_app;
-        }
-
-      while (capture_num)
-        {
-          ret = get_camimage(v_fd, &v4l2_buf, capture_type);
-          if (ret != OK)
-            {
-              goto exit_this_app;
-            }
-
-          futil_writeimage((uint8_t *)v4l2_buf.m.userptr,
-           (size_t)v4l2_buf.bytesused,
-           (capture_type == V4L2_BUF_TYPE_VIDEO_CAPTURE) ? "YUV" : "JPG");
+      is_eternal = 1;
+      printf("Start video this mode is eternal."
+             " (Non stop, non save files.)\n");
+#ifndef CONFIG_EXAMPLES_CAMERA_OUTPUT_LCD
+      printf("This mode should be run with LCD display\n");
+#endif
+    }
+  else
+    {
+      is_eternal = 0;
+      wait.tv_sec = START_CAPTURE_TIME;
+      wait.tv_usec = 0;
+      printf("Take %d pictures as %s file in %s after %d seconds.\n",
+             capture_num,
+              (capture_type == V4L2_BUF_TYPE_STILL_CAPTURE) ? "JPEG" : "RGB",
+             save_dir, START_CAPTURE_TIME);
+      printf(" After finishing taking pictures,"
+             " this app will be finished after %d seconds.\n",
+              KEEP_VIDEO_TIME);
+    }
 
-          ret = release_camimage(v_fd, &v4l2_buf);
-          if (ret != OK)
-            {
-              goto exit_this_app;
-            }
+  gettimeofday(&start, NULL);
 
-          capture_num--;
-        }
+  /* =====  Main Loop  ===== */
 
-      ret = stop_stillcapture(v_fd, capture_type);
-      if (ret != OK)
+  while (1)
+    {
+      switch (app_state)
         {
-          goto exit_this_app;
+          /* BEFORE_CAPTURE and AFTER_CAPTURE is waiting for expiring the
+           * time.
+           * In the meantime, Captureing VIDEO image to show pre-view on LCD.
+           */
+
+          case APP_STATE_BEFORE_CAPTURE:
+          case APP_STATE_AFTER_CAPTURE:
+            ret = get_camimage(v_fd, &v4l2_buf, V4L2_BUF_TYPE_VIDEO_CAPTURE);
+            if (ret != OK)
+              {
+                goto exit_this_app;
+              }
+
+#ifdef CONFIG_EXAMPLES_CAMERA_OUTPUT_LCD
+            nximage_draw((void *)v4l2_buf.m.userptr,
+                         VIDEO_HSIZE_QVGA, VIDEO_VSIZE_QVGA);
+#endif
+
+            ret = release_camimage(v_fd, &v4l2_buf);
+            if (ret != OK)
+              {
+                goto exit_this_app;
+              }
+
+            if (!is_eternal)
+              {
+                gettimeofday(&now, NULL);
+                timersub(&now, &start, &delta);
+                if (timercmp(&delta, &wait, >))
```

Comment:
@jerpelea 
The general implementation of `timercmp()` causes an nxstyle error. How can I fix this?

Reply:
-                if (timercmp(&delta, &wait, >))
+                if (timercmp(&delta,
+                             &wait,
+                             >
+                             ))

Reply:
```suggestion
-                if (timercmp(&delta, &wait, >))
+                if (timercmp(&delta,
+                             &wait,
+                             >
+                             ))

```

Reply:
@jerpelea thank you for your suggestion.
The readability is lower, but should I change to fix it as you suggested?

Reply:
Let's keep the current version and ignore style error.